pkg/endpoint/bpf.go

```diff
@@ -1229,6 +1273,17 @@ func (e *Endpoint) syncPolicyMap() error {
 			}
 		}
 
+		// Delete policy keys present in the realized state, but not present in the desired state
+		for keyToDelete := range e.realizedPolicy.PolicyDenyMapState {
+			// If key that is in realized state is not in desired state, just remove it.
+			if _, ok := e.desiredPolicy.PolicyDenyMapState[keyToDelete]; !ok {
+				if deleted, err := deletePolicyKey(e.policyDenyMap, e.realizedPolicy.PolicyDenyMapState, e.desiredPolicy.PolicyDenyMapState, keyToDelete, false, nil); !deleted {
+					e.getLogger().WithError(err).Error("Failed to delete PolicyMap key")
```

Comment:
update log message to signify deny case. Also maybe consider refactoring to remove code duplication?